Cargo.toml

```diff
@@ -17,6 +17,7 @@ rstest = "0.13"
 [dependencies]
 log = "0.4"
 cfg-if = "1"
+rustversion = "1.0.11"
```

Comment:
Agree. 

I added the dependency with cargo-add and didn't notice this. I will push a fix once I have access to my computer.

Reply:
Changed as suggested

Reply:
Yes, I have been working on adding tests these days. 

(To be exact, I am trying to set up a realtime kernel inside VM on my laptop, because I no longer have access to the realtime machine that I used before. )

I will let you know once I finish adding the tests.

Reply:
@vityafx I added several doctest cases for the scoped thread feature. This is tested on the following platform and all test cases pass. 

```sh
$ uname -a
Linux nixos 5.10.162-rt78 #1-NixOS SMP PREEMPT_RT Wed Jan 4 10:39:24 UTC 2023 x86_64 GNU/Linux
$ rustc --version
1.64.0
```

I also temporarily removed all the `no_run` annotations when testing, so I think all of the examples in the documentation are good.